ipol_demo/modules/core/archive/archive.py

```diff
@@ -0,0 +1,567 @@
+            message = "Failure in get_data_experiment. Error = {}".format(ex)
+            print(message)
+            self.logger.exception(message)
+            raise
+
+    def get_meta_info(self, conn, id_demo: int) -> list:
+        """
+        This function return the number of archive pages to be displayed
+        for a given demo, and the number of experiments done with
+        this demo.
+        """
+        meta_info = {}
+
+        meta_info[
+            "number_of_experiments_in_a_page"
+        ] = settings.number_of_experiments_by_pages
+
+        cursor_db = conn.cursor()
+
+        cursor_db.execute(
+            """
+        SELECT COUNT(*) FROM experiments WHERE id_demo = ?""",
+            (id_demo,),
+        )
+
+        number_of_experiments = cursor_db.fetchone()[0]
+        meta_info["number_of_experiments"] = number_of_experiments
+
+        if number_of_experiments == 0:
+            meta_info["first_date_of_an_experiment"] = "never"
+            meta_info["number_of_pages"] = 0
+            return meta_info
+        cursor_db.execute(
+            """
+        SELECT timestamp
+        FROM experiments WHERE id_demo = ?
+        ORDER BY timestamp """,
+            (id_demo,),
+        )
+
+        first_date_of_an_experiment = cursor_db.fetchone()[0]
+
+        if number_of_experiments % settings.number_of_experiments_by_pages != 0:
+            pages_to_add = 1
+        else:
+            pages_to_add = 0
+
+        number_of_pages = (
+            int(number_of_experiments / settings.number_of_experiments_by_pages)
+            + pages_to_add
+        )
+
+        meta_info["first_date_of_an_experiment"] = first_date_of_an_experiment
+        meta_info["number_of_pages"] = number_of_pages
+
+        return meta_info
+
+    def init_database(self) -> bool:
+        """
+        Initialize the database used by the module if it doesn't exist.
+        If the file is empty, the system delete it and create a new one.
+        """
+        if os.path.isfile(settings.database_file):
+            file_info = os.stat(settings.database_file)
+
+            if file_info.st_size == 0:
+                print(str(settings.database_file) + " is empty. Removing the file...")
+                try:
+                    self.logger.error("init_database: Database file was empty")
+                    os.remove(settings.database_file)
+                except Exception as ex:
+                    message = "Error in init_database. Error = {}".format(ex)
+                    self.logger.exception(message)
+                    return False
+
+                print("Creating a correct new database")
+
+        if not os.path.isfile(settings.database_file):
+            try:
+                conn = lite.connect(settings.database_file)
+                cursor_db = conn.cursor()
+
+                sql_buffer = ""
+
+                with open(
+                    settings.database_dir + "/drop_create_db_schema.sql", "r"
```

Comment:
```suggestion
               curdir = os.path.dirname(__file__)
                with open(
                    os.path.join(curdir, "drop_create_db_schema.sql"), "r"
```
(same as in demoinfo)
and remove the `database_dir` entry from the settings